client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDomServer from 'react-dom/server';
+import { connect } from 'react-redux';
+import { includes } from 'lodash';
+import tinymce from 'tinymce/tinymce';
+
+/**
+ * Internal dependencies
+ */
+import SuggestionList from './suggestion-list';
+import EditorMention from './editor-mention';
+import QueryUsersSuggestions from 'components/data/query-users-suggestions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getUserSuggestions } from 'state/users/suggestions/selectors';
+
+/**
+ * Module variables
+ */
+const VK = tinymce.util.VK;
+
+const getMatchingSuggestions = function( suggestions, query ) {
+	let matchingSuggestions = suggestions;
+
+	if ( query ) {
+		// Start of string or preceded by a space.
+		const matcher = new RegExp( '^' + query + '| ' + query, 'ig' );
+
+		matchingSuggestions = suggestions.filter( ( suggestion ) => {
+			const name = suggestion.name || suggestion.user_login + ' ' + suggestion.display_name;
+
+			return matcher.test( name );
+		} );
+	}
+
+	return matchingSuggestions.slice( 0, 10 );
+};
+
+export class Mentions extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			query: '',
+			popoverContext: null,
+			showPopover: false,
+		};
+	}
+
+	componentDidMount() {
+		const { editor } = this.props;
+		const position = this.getPosition();
+
+		editor.on( 'keyup', this.onKeyUp );
+		editor.on( 'click', this.hidePopover );
+
+		this.left = position.left;
+		this.top = position.top;
```

Comment:
another fun fact is that we can use destructuring too on the return from function calls!

```js
const { left, top } = this.getPosition();
this.left = left;
this.top = top;
```